contrib/test.sh

```diff
@@ -4,6 +4,19 @@ set -ex
 
 CRATES="bitcoin hashes internals fuzz"
 DEPS="recent minimal"
+MSRV="1\.48\.0"
+
+# Test pinned versions.
```

Comment:
```suggestion
# Test pinning works (this is the canonical list of required pins to build/test with MSRV).
```

Then we could add a section to the README pointing devs at this list? This could all be rolled into #1977 as a single commit.

Reply:
Since there is more to do I think It's ok to continue this as a separate PR.  Also from @apoelstra 

> As a separate PR we should add a test that directly tries to use the latest deps from cargo (and that will require the pins).

Reply:
Isn't that *this* PR?